Add english tutorial for loj-1249 : Chocolate thief (#358)
lightoj-dev/problem-tutorials: 1249/en.md

# LOJ 1249 - Chocolate Thief

### Problem statement
In this problem, each test case contains `n` number of lines. Each line contains:

>  `Name(n) length(l),breadth(b) height(h)`

Reviewer: add comma

> Denoting that the person with name(n) ate the cake of the length(l),breadth(b) & height. Each information separated by a `whitespace` .Each person should eat the same amount of cake `in volume` i.e each `l*b*h` should be the same.

Reviewer: ```suggestion
Denoting that the person with name (n) ate the cake of the length (l), breadth (b) and height (h). Each information is separated by a `whitespace`. Each person should eat the same amount (volume) of cake which means `l*b*h` should be the same.
```

> However, one single `thief` stole a portion from another `victim`, given that in all cases, 

Reviewer: ```suggestion
However, a single `thief` stole a portion from another `victim`, given that in all cases, 
```

```
number of thief = number of victim = 1
```

> We need to print out the `name` of the `thief` & `victim` in a particular testcase.

Reviewer: ```suggestion
We need to print out the `name` of the `thief` and `victim` in each testcase.
```

**Example Output:** `Case 1: mun took chocolate from all` 

### Approach

Basically we need to find out the name associated with `higher than average volume`  and `lower than average volume` 

> of consumed cake. I used this data-structure to solve the problem:

Reviewer: Please explain the structure a bit, basically, you are using a map to track name and frequency. I can see you added a comment on each line, but just write a paragraph saying the same. Also, its good to mention what this structure is as many people do not have an understanding of Javascript.

```js
const tab = {
    [key]: number // this denotes the volume of the consumed cake
    {
        name: string  // stored the name, not a necessary field in case of non-thiefs/non-victims
        freq: number // stored the number of people with the same volume i.e, the key
    }
}
```

> - If there is only single key, it means that everybody got the same volume of that cake

Reviewer: only one

> - Otherwise, we find out the **minimum** key and the **maximum** key, along with the associated **name**. Then print out the name of the **victim** and the **thief**

Reviewer: we need to find out

### Javascript
-----
```js
const readline = require('readline')
const rl = readline.createInterface({
    input:process.stdin
})
const userInput = []
rl.on('line',(values)=>{
    userInput.push(values)
})

rl.on('close',()=>{
    const noOfCases = parseInt(userInput[0])
    let inputPtr = 1
    for(let i=0;i<noOfCases;i++){
        let victim = ''
        let thief = ''

>         let cases = parseInt(userInput[inputPtr])

Reviewer: poor variable naming. should be `students`

        let tab = {}
        for(let i = inputPtr+1 ; i< cases+inputPtr+1;i++){

>             let temp = userInput[i].split(" ")

Reviewer: it would be better if you could split your code a bit. 
```
const l = parseInt(temp[temp.length-1]);
const h = ...
```

in this way code will be more readable.

            let vol = parseInt(temp[temp.length-1]) * parseInt(temp[temp.length-2]) * parseInt(temp[temp.length-3])
            let name = temp[0]
            if(tab[vol]){
                tab[vol].name = name
                tab[vol].freq++
            }
            else{
                tab[vol] = {
                    name,
                    freq:1
                }
            }
        }

>         let keys = Object.keys(tab)

Reviewer: though it doesn't matter but you should use `const` for this kind of variable than `let`
